### PR TITLE
feat: add snippets for all SDK components

### PR DESCRIPTION
**Pull request by rudouglas**
Changed the trigger to `NR1` to make them easier to find
Added Documentation links to each snippet
Updated the Readme

### CODE REVIEW

**Review by mbazhlekova — APPROVED** [review]
This is great! 🚀

File: snippets.json [hunk 1]
```diff
@@ -1,56 +1,1491 @@
...
+			"\tonChange={${5:this._onChange}}",
+			"\tspacingType={TextField.SPACING_TYPE.${6|EXTRA_LARGE,LARGE,MEDIUM,NONE,OMIT,SMALL|}}",
+			"\t${7:labelInline}",
+			"\t${8:disabled}",
+			"\t${9:autoFocus}",
+			"/>;"
+		],
+		"description": "TextField"
+	},
+	// Tables
```
Comment (zstix) [inline snippets.json:290]: Unfortunately, comments aren't supported in JSON.

```suggestion
```

File: snippets.json [hunk 2]
```diff
@@ -1,56 +1,1491 @@
...
+	"UserTableRowCell": {
+		"prefix": "NR1:UserTableRowCell",
+		"scope": "javascript,typescript",
+		"body": [
+			"// View more documentation at https://developer.newrelic.com/components/user-table-row-cell/",
+			"<UserTableRowCell value={${1:item}} />"
+		],
+		"description": "UserTableRowCell"
+	},
+	// Feedback
```
Comment (zstix) [inline snippets.json:418]:
```suggestion
```

File: snippets.json [hunk 3]
```diff
@@ -1,56 +1,1491 @@
...
+			"\t\t{",
+			"\t\t\tlabel: '${6:label}',",
+			"\t\t\tonClick: () => ${7:this._onClick()},",
+			"\t\t},",
+			"\t],",
+			"});"
+		],
+		"description": "Toast"
+	},
+	// Indicators
```
Comment (zstix) [inline snippets.json:504]:
```suggestion
```

File: snippets.json [hunk 4]
```diff
@@ -1,56 +1,1491 @@
...
+			"\t\tlabel: '${3:string}',",
+			"\t\tto: '${4:string}',",
+			"\t}}",
+			">",
+			"\t${5}",
+			"</Tooltip>"
+		],
+		"description": "Tooltip"
+	},
+	// Structure
```
Comment (zstix) [inline snippets.json:604]:
```suggestion
```

File: snippets.json [hunk 5]
```diff
@@ -1,56 +1,1491 @@
...
+			"\tto=\"${1:href}\"",
+			"\tspacingType={Link.SPACING_TYPE.${2|EXTRA_LARGE,LARGE,MEDIUM,NONE,OMIT,SMALL|}}",
+			"\tonClick={${3:this._onClick}}",
+			">",
+			"\t${4:label}",
+			"</Link>;"
+		],
+		"description": "Link"
+	},
+	// Charts
```
Comment (zstix) [inline snippets.json:868]:
```suggestion
```

File: snippets.json [hunk 6]
```diff
@@ -1,56 +1,1491 @@
...
+			"<TableChart",
+			"\taccountId={${1:accountId}}",
+			"\tquery=\"${2:NRQL}\"",
+			"\t${3:fullWidth}",
+			"\t${4:fullHeight}",
+			"/>;"
+		],
+		"description": "TableChart"
+	},
+	// Query and Storage
```
Comment (zstix) [inline snippets.json:1060]:
```suggestion
```

**Review by zstix — CHANGES_REQUESTED** [review]
Awesome work!

The only feedback I have is that the JS-style comments (`//`) aren't supported in JSON and might break whatever system attempts to consume this file.

File: snippets.json (hunk 1 above)
Comment (aswanson-nr) [reply snippets.json:290]: VSCode allows C-style comments within snippet files! https://code.visualstudio.com/docs/editor/userdefinedsnippets#_create-your-own-snippets

**Review by aswanson-nr — APPROVED** [review]
LGTM! You should be able to resolve @zstix comments since VSCode allows those inline code comments.

**Review by zstix — APPROVED** [review]

**Pull request merged**